src/devices/ovrheadset/OVRHeadset.cpp

```diff
@@ -852,7 +885,9 @@ void yarp::dev::OVRHeadset::run()
     yarp::os::Stamp predicted_stamp(distortionFrameIndex, predicted_ts.HeadPose.TimeInSeconds);
 
     // Get Input State
-    ovr_GetInputState(session, ovrControllerType_Active, &inputState);
+    inputStateMutex.lock();
+    result = ovr_GetInputState(session, ovrControllerType_Active, &inputState);
```

Comment:
I wonder if putting this blocking call on a separated thread may improve performance of this display thread?